### PR TITLE
Add and organise tests for code blocks

### LINKED ISSUE #1144: Add tests for markdown-it
**Edit: I now feel that that's no real need to apply snapshot testing as the `test-sites` are essentially a way of snapshot testing. If you run tests (i.e. `npm run test`/`npm run testwin`) and some other MarkBind related feature shows up as a diff, the developer can then infer if markdown-it is the cause of it, depending on the changes he made.** 

**Having said this, I feel that it is then more worthwhile to ensure that the test suite we have covers syntax and features rendered by MarkBind.**

**Is your request related to a problem?**

The scope of code the `test_sites` cover encompasses the entire build functionality of MarkBind, although you can argue that 'only the relevant pieces of code play a role in rendering the output of the test site html'. 

However, I believe it would be good practice to test on a more narrower scope. If something goes wrong in the test site output, how do we know whether or not it's a problem with our parsers?

**Describe the solution you'd like**

Therefore, as a first step to making more smaller unit tests, we should make tests on markdown-it via jest.

**Describe alternatives you've considered**

Not make tests for markdown-it and just leave it all to `test_sites`

**Additional context**

Since it would be overly tedious to manually type the expected output of MarkBind/Markdown syntax, jest snapshot testing is used, which you can think of as a 'unit test with an auto generated assumption about your component.' . https://stackoverflow.com/questions/43771602/jest-snapshot-testing-vs-unit-testing

### PR DESCRIPTION
**Pull request by nbriannl**
**What is the purpose of this pull request? (put "X" next to an item, remove the rest)**

• [ ] Documentation update
• [ ] Bug fix
• [ ] New feature
• [ ] Enhancement to an existing feature
• [X] Other, please explain: Testing

**What is the rationale for this request?**

Added more tests than will more comprehensively cover the behaviors of components I've worked on. Particular ~Boxes and~ Code Block, since I've done PRs ~https://github.com/MarkBind/markbind/pull/1034, https://github.com/MarkBind/markbind/pull/1137, https://github.com/MarkBind/vue-strap/pull/129~

**What changes did you make? (Give an overview)**

* Added test for Code Block
* ~Added test for Boxes, that are related to MarkBind parsing,~
  * ~The test suite cannot cover VueStrap rendering. VueStrap tests are done via https://github.com/MarkBind/vue-strap/pull/146~

**Is there anything you'd like reviewers to focus on?**

Not sure other than naming of tests and any coverage I've left out. 

**Testing instructions:**

Check and run tests.

**Proposed commit message: (wrap lines at 72 characters)**

Add and organise tests for code blocks

### CODE REVIEW

File: test/unit/markdown-it/index.test.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
+const markdownIt = require('../../../src/lib/markbind/src/lib/markdown-it/');
+
+// test('renders 1', () => {
```
Comment (openorclose) [inline test/unit/markdown-it/index.test.js:3]: Why are these commented out?

If you want, you can use `test.skip` instead to skip unwanted tests.

File: test/unit/markdown-it/index.test.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+// });
+
+// test('renders 2', () => {
+//   const result = markdownIt.render('# heading');
+//   const expected = '<h1>heading</h1>\n';
+//   expect(result).toEqual(expected);
+// });
+
+test('markdown-it renders a fenced code block', () => {
+  const source = [
```
Comment (openorclose) [inline test/unit/markdown-it/index.test.js:16]: Should probably add a comment to explain why you had to split up the lines... does eslint complain if you just do

````
   const source = `\`\`\`xml
some lines
some lines
```;
````

File: test/unit/markdown-it/index.test.js (hunk 1 above)
Comment (nbriannl) [reply test/unit/markdown-it/index.test.js:3]: This PR is still WIP. 😅 I planned to delete this.

Comment (openorclose) [reply test/unit/markdown-it/index.test.js:3]: Oh oops didn't see the wip

File: test/functional/test_site/testBoxes.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
+**Test: Boxes**
```
Comment (ang-zeyu) [inline test/functional/test_site/testBoxes.md:1]: Definitely see the purpose of the code block tests, what does this cover that `componentParserTest` misses though?

Comment (nbriannl) [reply test/functional/test_site/testBoxes.md:1]: Ah, does componentParserTest cover the rendering of markdown inside components?

I'll go check later.

Comment (nbriannl) [reply test/functional/test_site/testBoxes.md:1]: Ah yeah, I found that it covered. My bad 👍

Comment (ang-zeyu) [reply test/functional/test_site/testBoxes.md]: ok, the code blocks tests looks good 👍

**Pull request merged**